flytekit/core/type_engine.py

```diff
@@ -196,6 +188,35 @@ def to_html(self, ctx: FlyteContext, python_val: T, expected_python_type: Type[T
         """
         return str(python_val)
 
+    def to_json(self, ctx: FlyteContext, python_val: T, python_type: Type[T], expected: LiteralType) -> Literal:
+        json_str = json.dumps(python_val)
+        json_bytes = json_str.encode("UTF-8")
+        return Literal(scalar=Scalar(json=Json(value=json_bytes, serialization_format="UTF-8")))
+
+    def from_json(self, ctx: FlyteContext, json_idl_object: Json, expected_python_type: Type[T]) -> T:
+        if expected_python_type in [datetime.datetime, datetime.timedelta, datetime.date]:
+            raise TypeTransformerFailedError(
+                f"Unsupported Type Error: JSON IDL serialization/deserialization is not supported for Python type "
+                f"'{expected_python_type.__name__}'.\n"
+                f"Please ensure that the type is serializable or convert it to a supported format."
+            )
+
+        value = json_idl_object.value
+        serialization_format = json_idl_object.serialization_format
+        if serialization_format == "UTF-8":
+            json_str = value.decode("UTF-8")
+        else:
+            raise ValueError(
+                f"Bytes can't be converted to JSON String.\n"
+                f"Unsupported serialization format: {serialization_format}"
+            )
+        python_val =  json.loads(json_str)
+        expected_python_val = expected_python_type(python_val)
```

Comment:
In golang, if we use `json.Unmarshal`, it will discard `.0`, which will make `json.loads(float)` to int, if the value is `x.0` (for example `2.0`).

<img width="378" alt="image" src="https://github.com/user-attachments/assets/99f3a521-f1c4-438e-98d1-1f0318c0e6ee">